prow/github/types.go

```diff
@@ -60,6 +60,12 @@ type PullRequestBranch struct {
 	Repo Repo   `json:"repo"`
 }
 
+type Label struct {
```

Comment:
It's a straightforward rest api, and I needed some lower-level control than the go-github client provided. At one point I was using my own client but the types in go-github, but I don't think it was worth it.

I also roll my own kubernetes client, because when I wrote this the client-go repo didn't exist. Might be worth transitioning.